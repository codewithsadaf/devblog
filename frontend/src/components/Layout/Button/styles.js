//imports
import styled, { css } from "styled-components";

const handleButtonVariant = {
    default: (theme) => css`
        background: ${theme.colors.primaryColor};
        color: ${theme.colors.white};
        border: 2px solid ${theme.colors.primaryColor};
        border-radius: 4px;
        transition: ${theme.transition.tran_03};

        &:hover {
            background: transparent;
            color: ${theme.colors.primaryColor};
            border: 2px solid ${theme.colors.primaryColor};
            transition: ${theme.transition.tran_03};
        }
    `,
    outline: (theme) => css`
        background: transparent;
        color: ${theme.colors.primaryColor};
        border: 2px solid ${theme.colors.primaryColor};
        border-radius: 4px;
        transition: ${theme.transition.tran_03};

        &:hover {
            background: ${theme.colors.primaryColor};
            color: ${theme.colors.white};
            border: 2px solid ${theme.colors.primaryColor};
            transition: ${theme.transition.tran_03};
        }
    `,

    success: (theme) => css`
        background: ${theme.colors.success};
        color: ${theme.colors.white};
        border: 2px solid ${theme.colors.success};
        border-radius: 4px;
        transition: ${theme.transition.tran_03};

        &:hover {
            opacity: 0.8;
            transition: ${theme.transition.tran_03};
        }
    `,

    danger: (theme) => css`
        background: ${theme.colors.danger};
        color: ${theme.colors.white};
        border: 2px solid ${theme.colors.danger};
        border-radius: 4px;
        transition: ${theme.transition.tran_03};

        &:hover {
            opacity: 0.8;
            transition: ${theme.transition.tran_03};
        }
    `,
};

const handleUppercase = () => css`
    text-transform: uppercase;
`;

const handleMaxWidth = () => css`
    width: 100%;
    padding: 1em 0;
`;

const handleSmallStyles = (theme) => css`
    padding: 7px 30px;
    border-radius: 4px;
    font-size: ${theme.font.sizes.small};
    border: 1px solid;

    &:hover {
        border: 1px solid;
    }
`;

export const ButtonContainer = styled.button`
    ${({ theme, buttonVariant, uppercase, maxWidth, isSmall }) => css`
        padding: 0.5em 1.8em;
        font-size: 1em;
        border: none;
        font-weight: bold;
        cursor: pointer;
        ${handleButtonVariant[buttonVariant](theme)};
        ${uppercase && handleUppercase()};
        ${maxWidth && handleMaxWidth()};
        ${isSmall && handleSmallStyles(theme)};
    `}
`;
