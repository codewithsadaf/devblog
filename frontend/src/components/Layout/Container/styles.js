//imports
import styled from "styled-components";

export const Container = styled.div`
    min-height: 60vh;
    margin-bottom: 5em;
    padding: 0 2em;
`;
