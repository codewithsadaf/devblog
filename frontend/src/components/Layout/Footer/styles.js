//imports
import styled, { css } from "styled-components";

export const FooterWrapper = styled.footer`
    ${({ theme }) => css`
        height: 250px;
        width: 100%;
        display: flex;
        align-items: center;
        justify-content: center;
        flex-direction: column;
        background-color: ${theme.colors.footerBackground};
        text-align: center;
    `}
`;
export const FooterHeading = styled.h3`
    ${({ theme }) => css`
        @media ${theme.media.lteMedium} {
            font-size: ${theme.font.sizes.medium};
            line-height: ${theme.font.line_height.h6};
        }
    `}
`;

export const CopyrightText = styled.span`
    display: block;
    margin-top: 0.8em;
`;
