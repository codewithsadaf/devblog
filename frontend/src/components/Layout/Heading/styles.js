//imports
import styled, { css } from "styled-components";

const headingSize = {
    huge: (theme) => css`
        font-size: ${theme.font.sizes.xhuge};
        line-height: ${theme.font.line_height.h1};
        ${mediaFont(theme)};
    `,
    big: (theme) => css`
        font-size: ${theme.font.sizes.huge};
        line-height: ${theme.font.line_height.h2};
    `,
    large: (theme) => css`
        font-size: ${theme.font.sizes.xxlarge};
        line-height: ${theme.font.line_height.h3};
    `,
    medium: (theme) => css`
        font-size: ${theme.font.sizes.medium};
        line-height: ${theme.font.line_height.h6};
    `,
    small: (theme) => css`
        font-size: ${theme.font.sizes.small};
        line-height: ${theme.font.line_height.small};
    `,
    xsmall: (theme) => css`
        font-size: ${theme.font.sizes.xsmall};
        line-height: ${theme.font.line_height.small};
    `,
};

const mediaFont = (theme) => css`
    @media ${theme.media.lteMedium} {
        font-size: ${theme.font.sizes.xlarge};
        line-height: ${theme.font.line_height.h4};
    }
`;

export const Container = styled.h1`
    ${({ theme, size, uppercase, bold }) => css`
        text-transform: ${uppercase ? "uppercase" : "none"};
        font-weight: ${bold ? "bold" : 500};
        ${headingSize[size](theme)};
    `}
`;
