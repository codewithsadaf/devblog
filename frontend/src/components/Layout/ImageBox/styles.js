//imports
import styled, { css } from "styled-components";

const handleBigStyle = () => css`
    width: 100%;
    height: auto;
`;

export const Container = styled.div`
    ${({ theme }) => css``}
`;

export const ImageElm = styled.img`
    ${({ rounded }) => css`
        width: 150px;
        height: 150px;
        object-fit: cover;
        display: block;
        margin: 1em auto;
        border-radius: ${rounded ? "50%" : "7px"};
    `}
`;
