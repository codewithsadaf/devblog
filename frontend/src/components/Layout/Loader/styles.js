//imports
import styled, { css } from "styled-components";

export const Container = styled.div`
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    display: flex;
    align-items: center;
    justify-content: center;
    width: 100%;
    height: 100vh;
    background: rgba(0, 0, 0, 0.6);
    z-index: 99999;
    pointer-events: all;
`;

export const Circle = styled.div`
    ${({ theme }) => css`
        &,
        &::before,
        &::after {
            border-radius: 50%;
            width: 2.5em;
            height: 2.5em;
            animation-fill-mode: both;
            animation: bblFadInOut 1.8s infinite ease-in-out;
        }

        & {
            color: ${theme.colors.white};
            font-size: 7px;
            position: relative;
            text-indent: -9999em;
            transform: translateZ(0);
            animation-delay: -0.16s;
        }

        &::before,
        &:after {
            content: "";
            position: absolute;
            top: 0;
        }

        &::before {
            left: -3.5em;
            animation-delay: -0.32s;
        }

        &::after {
            left: 3.5em;
        }

        @keyframes bblFadInOut {
            0%,
            80%,
            100% {
                box-shadow: 0 2.5em 0 -1.3em;
            }
            40% {
                box-shadow: 0 2.5em 0 0;
            }
        }
    `}
`;
