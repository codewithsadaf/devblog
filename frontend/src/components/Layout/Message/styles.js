//imports
import styled, { css } from "styled-components";
//react icons
import { IoClose } from "react-icons/io5";

export const Container = styled.div`
    ${({ theme }) => css`
        display: flex;
        align-items: center;
        justify-content: space-between;
        position: fixed;
        bottom: 30px;
        left: 50%;
        transform: translateX(-50%);
        padding: 25px 30px;
        min-width: 25%;
        border-radius: 7px;
        z-index: 9999;
        font-weight: bold;
        box-shadow: rgba(0, 0, 0, 0.24) 0px 3px 8px;
        transition: ${theme.transition.tran_03};

        &.success {
            background-color: ${theme.colors.success};
            color: ${theme.colors.white};
        }

        &.error {
            background-color: ${theme.colors.danger};
            color: ${theme.colors.white};
        }

        @media ${theme.media.lteMedium} {
            bottom: 0;
            right: 50%;
            left: 50%;
            min-width: 90%;
            transform: translate(-50%, -50%);
        }
    `}
`;

export const CloseIcon = styled(IoClose)`
    ${({ theme }) => css`
        color: ${theme.colors.white};
        cursor: pointer;
        font-size: 19px;
        margin-left: 0.3em;
    `}
`;

export const MessageText = styled.span`
    display: block;
`;
