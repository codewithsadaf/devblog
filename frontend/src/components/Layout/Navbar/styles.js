//imports
import { Link } from "react-router-dom";
import styled, { css } from "styled-components";
//react icons
import { MdClose } from "react-icons/md";
import { GiHamburgerMenu } from "react-icons/gi";

export const Nav = styled.nav`
    display: flex;
    align-items: center;
    justify-content: space-between;
    width: 100%;
    height: 70px;
    padding: 1.2em 2em;
    box-shadow: 0 2px 10px 0 rgba(0, 0, 0, 0.15);
    position: relative;
`;

export const Nav_brand = styled(Link)`
    ${({ theme }) => css`
        font-size: ${theme.font.sizes.large};
        color: ${theme.colors.primaryColor};
    `}
`;

export const bold_Effect = styled.span`
    font-weight: 900;
    text-transform: uppercase;
`;

export const Menu = styled.ul`
    ${({ theme, open }) => css`
        display: flex;

        @media ${theme.media.lteMedium} {
            position: fixed;
            top: 0;
            right: ${open ? "0" : "-300px"};
            height: 100vh;
            width: 300px;
            display: flex;
            flex-direction: column;
            background: ${theme.colors.primaryColor};
            padding: 30px;
            transition: ${theme.transition.tran_05};
            z-index: 999999;
        }
    `}
`;

export const List_Item = styled.li`
    ${({ theme }) => css`
        margin-left: ${theme.spacings.medium};

        @media ${theme.media.lteMedium} {
            margin-left: 0;
            margin-top: ${theme.spacings.medium};
        }
    `}
`;

export const HamburgerIcon = styled(GiHamburgerMenu)`
    ${({ theme }) => css`
        font-size: 26px;
        display: none;
        visibility: hidden;
        opacity: 0;
        pointer-events: none;
        cursor: pointer;

        @media ${theme.media.lteMedium} {
            display: block;
            visibility: visible;
            opacity: 1;
            pointer-events: all;
        }
    `}
`;

export const CloseIcon = styled(MdClose)`
    ${({ theme }) => css`
        font-size: 26px;
        display: none;
        visibility: hidden;
        opacity: 0;
        pointer-events: none;
        cursor: pointer;
        color: ${theme.colors.white};

        @media ${theme.media.lteMedium} {
            display: block;
            visibility: visible;
            opacity: 1;
            pointer-events: all;
            border-radius: 50%;
        }
    `}
`;

export const LogoutText = styled.span`
    ${({ theme }) => css`
        font-size: ${theme.font.sizes.default};
        padding: 0.4em 0.5em;
        color: ${theme.colors.primaryColor};
        font-weight: 500;
        transition: ${theme.transition.tran_03};
        text-transform: capitalize;
        cursor: pointer;

        &:hover {
            opacity: 0.8;
            transition: ${theme.transition.tran_03};
        }

        @media ${theme.media.lteMedium} {
            color: ${theme.colors.white};
            font-size: ${theme.font.sizes.medium};
        }
    `}
`;
