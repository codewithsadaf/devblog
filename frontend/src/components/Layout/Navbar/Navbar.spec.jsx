//imports
import { fireEvent, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
//styles
import { renderTheme } from "../../../styles/render-theme";
import { theme } from "../../../styles/theme";
//contexts
import { Context as UserContext } from "../../../context/UserContext";
//components
import { Navbar } from "./";

let logoutMock = jest.fn();

describe("<Navbar />", () => {
    describe("Rendering", () => {
        beforeEach(() => {
            renderTheme(
                <BrowserRouter>
                    <UserContext.Provider
                        value={{
                            authenticated: false,
                            logout: logoutMock,
                        }}
                    >
                        <Navbar />
                    </UserContext.Provider>
                </BrowserRouter>
            );
        });

        it("should render Navbar", () => {
            const heading = screen.getByRole("link", { name: "Dev Blog" });

            expect(heading).toBeInTheDocument();
        });

        it("should render Navbar with menu hidden when screen is less then 720px", () => {
            expect(
                screen.getByLabelText("Close Icon").parentElement
            ).toHaveStyleRule("right", "-300px", {
                media: theme.media.lteMedium,
            });
        });
    });

    describe("Menu functionality", () => {
        beforeEach(() => {
            renderTheme(
                <BrowserRouter>
                    <UserContext.Provider
                        value={{
                            authenticated: false,
                            logout: logoutMock,
                        }}
                    >
                        <Navbar />
                    </UserContext.Provider>
                </BrowserRouter>
            );
        });

        it("should open menu when hamburger icon is clicked on mobile", () => {
            const menuElem = screen.getByLabelText("Close Icon").parentElement;

            expect(menuElem).toHaveStyleRule("right", "-300px", {
                media: theme.media.lteMedium,
            });

            const hamburgerIcon = screen.getByLabelText("Hamburger Icon");

            fireEvent.click(hamburgerIcon);

            expect(menuElem).toHaveStyleRule("right", "0", {
                media: theme.media.lteMedium,
            });
        });

        it("should close menu when close icon is clicked on mobile", () => {
            const menuElem = screen.getByLabelText("Close Icon").parentElement;
            const hamburgerIcon = screen.getByLabelText("Hamburger Icon");
            const closeIcon = screen.getByLabelText("Close Icon");

            expect(menuElem).toHaveStyleRule("right", "-300px", {
                media: theme.media.lteMedium,
            });

            fireEvent.click(hamburgerIcon);

            expect(menuElem).toHaveStyleRule("right", "0", {
                media: theme.media.lteMedium,
            });

            fireEvent.click(closeIcon);

            expect(menuElem).toHaveStyleRule("right", "-300px", {
                media: theme.media.lteMedium,
            });
        });
    });

    describe("Authenticated functionality", () => {
        beforeEach(() => {
            logoutMock = jest.fn();
        });

        afterEach(() => {
            logoutMock.mockClear();
        });

        it("should show authenticated links when user is authenticated", () => {
            renderTheme(
                <BrowserRouter>
                    <UserContext.Provider
                        value={{ authenticated: true, logout: logoutMock }}
                    >
                        <Navbar />
                    </UserContext.Provider>
                </BrowserRouter>
            );

            const newPostLink = screen.getByRole("link", {
                name: /Novo post/i,
            });
            const dashboardLink = screen.getByRole("link", {
                name: /Dashboard/i,
            });

            const logOutLink = screen.getByText(/Sair/i);

            expect(newPostLink).toBeInTheDocument();
            expect(dashboardLink).toBeInTheDocument();
            expect(logOutLink).toBeInTheDocument();
        });

        it("should call logout function on logout span click and close menuBar on mobile", () => {
            renderTheme(
                <BrowserRouter>
                    <UserContext.Provider
                        value={{ authenticated: true, logout: logoutMock }}
                    >
                        <Navbar />
                    </UserContext.Provider>
                </BrowserRouter>
            );

            const menuElem = screen.getByLabelText("Close Icon").parentElement;
            const hamburgerIcon = screen.getByLabelText("Hamburger Icon");
            const logOutLink = screen.getByText(/Sair/i);

            expect(menuElem).toHaveStyleRule("right", "-300px", {
                media: theme.media.lteMedium,
            });

            fireEvent.click(hamburgerIcon);

            expect(menuElem).toHaveStyleRule("right", "0", {
                media: theme.media.lteMedium,
            });

            fireEvent.click(logOutLink);

            expect(logoutMock).toHaveBeenCalledTimes(1);
            expect(menuElem).toHaveStyleRule("right", "-300px", {
                media: theme.media.lteMedium,
            });
        });

        it("should call logout function on logout span click", () => {
            renderTheme(
                <BrowserRouter>
                    <UserContext.Provider
                        value={{ authenticated: true, logout: logoutMock }}
                    >
                        <Navbar />
                    </UserContext.Provider>
                </BrowserRouter>
            );

            const logOutLink = screen.getByText(/Sair/i);

            fireEvent.click(logOutLink);

            expect(logoutMock).toHaveBeenCalledTimes(1);
        });

        it("should hide authenticated links when user is not authenticated", () => {
            renderTheme(
                <BrowserRouter>
                    <UserContext.Provider
                        value={{ authenticated: false, logout: logoutMock }}
                    >
                        <Navbar />
                    </UserContext.Provider>
                </BrowserRouter>
            );

            const newPostLink = screen.queryByRole("link", {
                name: /Novo post/i,
            });
            const dashboardLink = screen.queryByRole("link", {
                name: /Dashboard/i,
            });

            const logOutLink = screen.queryByText(/Sair/i);

            expect(newPostLink).not.toBeInTheDocument();
            expect(dashboardLink).not.toBeInTheDocument();
            expect(logOutLink).not.toBeInTheDocument();
        });
    });

    describe("Snapshot", () => {
        it("should match snapshot", () => {
            const { container } = renderTheme(
                <BrowserRouter>
                    <UserContext.Provider
                        value={{ authenticated: false, logout: logoutMock }}
                    >
                        <Navbar />
                    </UserContext.Provider>
                </BrowserRouter>
            );

            expect(container.firstChild).toMatchSnapshot();
        });
    });
});
