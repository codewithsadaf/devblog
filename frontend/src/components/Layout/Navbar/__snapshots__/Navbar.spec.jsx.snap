// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> Snapshot should match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
  height: 70px;
  padding: 1.2em 2em;
  box-shadow: 0 2px 10px 0 rgba(0,0,0,0.15);
  position: relative;
}

.c1 {
  font-size: 1.4em;
  color: #000;
}

.c2 {
  font-weight: 900;
  text-transform: uppercase;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c5 {
  margin-left: 1.2em;
}

.c7 {
  font-size: 26px;
  display: none;
  visibility: hidden;
  opacity: 0;
  pointer-events: none;
  cursor: pointer;
}

.c4 {
  font-size: 26px;
  display: none;
  visibility: hidden;
  opacity: 0;
  pointer-events: none;
  cursor: pointer;
  color: #fff;
}

.c6 {
  font-size: 1em;
  padding: 0.4em 0.5em;
  color: #000;
  font-weight: 500;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
  text-transform: capitalize;
}

.c6:hover {
  opacity: 0.8;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c6.active {
  background-color: #000;
  color: #fff;
  border-radius: 3px;
  cursor: default;
}

.c6.active:hover {
  opacity: 1;
}

@media (max-width:768px) {
  .c3 {
    position: fixed;
    top: 0;
    right: -300px;
    height: 100vh;
    width: 300px;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    background: #000;
    padding: 30px;
    -webkit-transition: all 500ms ease-in-out;
    transition: all 500ms ease-in-out;
    z-index: 999999;
  }
}

@media (max-width:768px) {
  .c5 {
    margin-left: 0;
    margin-top: 1.2em;
  }
}

@media (max-width:768px) {
  .c7 {
    display: block;
    visibility: visible;
    opacity: 1;
    pointer-events: all;
  }
}

@media (max-width:768px) {
  .c4 {
    display: block;
    visibility: visible;
    opacity: 1;
    pointer-events: all;
    border-radius: 50%;
  }
}

@media (max-width:768px) {
  .c6.active {
    cursor: auto;
  }
}

@media (max-width:768px) {
  .c6 {
    color: #fff;
    font-size: 1.2em;
  }
}

<nav
  class="c0"
>
  <a
    class="c1"
    href="/"
  >
    Dev
    <span
      class="c2"
    >
      Blog
    </span>
  </a>
  <ul
    class="c3"
  >
    <svg
      aria-label="Close Icon"
      class="c4"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
      />
    </svg>
    <li
      class="c5"
    >
      <a
        class="c6"
        href="/home"
      >
        Home
      </a>
    </li>
    <li
      class="c5"
    >
      <a
        class="c6"
        href="/login"
      >
        Entrar
      </a>
    </li>
    <li
      class="c5"
    >
      <a
        class="c6"
        href="/register"
      >
        Cadastrar
      </a>
    </li>
    <li
      class="c5"
    >
      <a
        class="c6"
        href="/about"
      >
        Sobre
      </a>
    </li>
  </ul>
  <svg
    aria-label="Hamburger Icon"
    class="c7"
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    stroke-width="0"
    viewBox="0 0 512 512"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M32 96v64h448V96H32zm0 128v64h448v-64H32zm0 128v64h448v-64H32z"
    />
  </svg>
</nav>
`;
